lib/filewatch/ruby_fnv.rb

```diff
@@ -0,0 +1,63 @@
+# fnv code extracted and modified from https://github.com/jakedouglas/fnv-ruby
```

Comment:
this has perhaps been discussed in depth somewhere else, but can we document why we decided to use this kind of hashing algorithm vs sha1 or other known ones?